lib/knex-builder/make-knex.js

```diff
@@ -123,6 +123,12 @@ function initContext(knexFn) {
     // If container is not provided, returns a promise with a transaction that is resolved
     // when transaction is ready to be used.
     transaction(container, _config) {
+      // Overload support of `transaction(config)`
+      if (!_config && typeof container === 'object') {
```

Comment:
It would be safer to use method `isObject` from `lib/util/is.js`, because null is also an object.